### PR TITLE
Move to Go 1.9

### PR DESCRIPTION
**Pull request by daviddrysdale**
Prob'ly best reviewed commit-by-commit.

### CODE REVIEW

File: jsonclient/client.go [hunk 1]
```diff
@@ -127,9 +127,6 @@ func New(uri string, hc *http.Client, opts Options) (*JSONClient, error) {
 // returned http.Response can be non-nil even when an error is returned,
 // in particular when the HTTP status is not OK or when the JSON parsing fails.
 func (c *JSONClient) GetAndParse(ctx context.Context, path string, params map[string]string, rsp interface{}) (*http.Response, []byte, error) {
-	if ctx == nil {
```
Comment (phad) [inline jsonclient/client.go]: This is fine, but not sure the commit belongs in this PR?

File: trillian/ctfe/handlers_test.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ import (
 
 	"github.com/golang/glog"
 	"github.com/golang/mock/gomock"
-	"github.com/google/certificate-transparency-go"
+	ct "github.com/google/certificate-transparency-go"
```
Comment (phad) [inline trillian/ctfe/handlers_test.go]: The import alias 'ct' isn't being used in other diffs for this commit ...  is the import necessary at all?

Comment (daviddrysdale) [reply trillian/ctfe/handlers_test.go]: Yes; the code uses `ct`, which is the top-level package name (i.e. it doesn't match the last path component).  This makes it explicit, which helps tools like `gofmt`.

File: jsonclient/client.go (hunk 1 above)
Comment (daviddrysdale) [reply jsonclient/client.go]: dropped

File: trillian/ctfe/handlers_test.go (hunk 2 above)
Comment (phad) [reply trillian/ctfe/handlers_test.go]: OK, couldn't see it easily in the GH code review page.

**Pull request merged**